src/pool/mempool.rs

```diff
@@ -338,7 +352,31 @@ where
                         }
 
                         let sealed_block = latest_block.seal(hash);
-                        let mined_transactions = sealed_block.body.transactions.iter().map(|tx| tx.hash).collect();
+                        let mut mined_transactions: Vec<_> =
+                            sealed_block.body.transactions.iter().map(|tx| tx.hash).collect();
+
+                        // Prune mined transactions from the mempool mapping
+                        for tx_hash in &mined_transactions {
+                            mempool_transactions.remove(tx_hash);
+                        }
+
+                        // Prune transactions that have been in the mempool for more than 5 minutes
+                        let now = Instant::now();
+
+                        for (tx_hash, timestamp) in &mempool_transactions.clone() {
```

Comment:
do we need the `clone()`?

Reply:
ah ok but then we can just do `mempool_transactions.clone()` no? no need for the reference I believe.

Reply:
yes we can just modified it, this just induced some inverting some "take by value" or "take by reference" in the loop but this is cleaner :)